src/cluster_legacy.c

```diff
@@ -884,38 +939,37 @@ void clusterUpdateMyselfIp(void) {
     }
 }
 
-/* Update the hostname for the specified node with the provided C string. */
-static void updateAnnouncedHostname(clusterNode *node, char *new) {
-    /* Previous and new hostname are the same, no need to update. */
-    if (new && !strcmp(new, node->hostname)) {
+static void updateSdsExtensionField(char **field, const char *value) {
+    if (value != NULL && !strcmp(value, *field)) {
         return;
-    } else if (!new && (sdslen(node->hostname) == 0)) {
+    } else if (value == NULL && sdslen(*field) == 0) {
         return;
     }
 
-    if (new) {
-        node->hostname = sdscpy(node->hostname, new);
-    } else if (sdslen(node->hostname) != 0) {
-        sdsclear(node->hostname);
+    if (value != NULL) {
+        *field = sdscpy(*field, value);
+    } else {
+        sdsclear(*field);
     }
     clusterDoBeforeSleep(CLUSTER_TODO_SAVE_CONFIG);
 }
 
-static void updateAnnouncedHumanNodename(clusterNode *node, char *new) {
-    if (new && !strcmp(new, node->human_nodename)) {
-        return;
-    } else if (!new && (sdslen(node->human_nodename) == 0)) {
-        return;
-    }
+/* Update the hostname for the specified node with the provided C string. */
+static void updateAnnouncedHostname(clusterNode *node, char *value) {
```

Comment:
looks like `value` is always an `sds` in all call sites? 
```suggestion
static void updateAnnouncedHostname(clusterNode *node, sds value) {
```

Reply:
No, not when it comes from the cluster bus. In clusterProcessPingExtensions we have

```C
    char *ext_hostname = NULL;
    ...
            clusterMsgPingExtHostname *hostname_ext = (clusterMsgPingExtHostname *)&(ext->ext[0].hostname);
            ext_hostname = hostname_ext->hostname;
    ...
    updateAnnouncedHostname(sender, ext_hostname);
```